### PR TITLE
Add support for context

### PR DESCRIPTION
**Pull request by qzaidi**
I have added a RunWithContext function for most ops, and we are using this in production, so hopefully should be fine.

### CODE REVIEW

File: substitute.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/dynamodb"
 
-	"github.com/guregu/dynamo/internal/exprs"
+	"github.com/qzaidi/dynamo/internal/exprs"
```
Comment (guregu) [inline substitute.go]: Could you fix this?

**Comment by guregu** [comment]
Thanks. It looks good to me.

**Comment by eafzali** [comment]
Hey! are you going to merge this soon @guregu?

**Comment by guregu** [comment]
Hello, sorry for the wait. There were lots of other bits of the package that needed to be made context-aware, specifically the retry functionality and batch iterators. 

I've given it a shot at the `ctx` branch, check it out https://github.com/guregu/dynamo/compare/ctx.
Let me know if it looks good I'll merge it all. 

cc @eafzali @qzaidi @edwardssb

**Pull request merged**